Runtime/GameLiftCoreApi.cs

```diff
@@ -3,51 +3,51 @@
 
 #if !UNITY_SERVER
 
-using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Amazon.GameLift;
+using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core;
 using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
 using AmazonGameLiftPlugin.Core.ApiGatewayManagement.Models;
 using AmazonGameLiftPlugin.Core.Latency;
 using AmazonGameLiftPlugin.Core.Latency.Models;
 using AmazonGameLiftPlugin.Core.Shared;
 using AmazonGameLiftPlugin.Core.UserIdentityManagement;
 using AmazonGameLiftPlugin.Core.UserIdentityManagement.Models;
+using UnityEngine;
 
 namespace AmazonGameLift.Runtime
 {
     public class GameLiftCoreApi
     {
         private readonly GameLiftConfiguration _configuration;
-        private readonly bool _isLocalMode;
+        private readonly bool _isAnywhereMode;
+        private CoreApi _coreApi;
 
-        public GameLiftCoreApi(GameLiftConfiguration configuration)
+        protected GameLiftCoreApi(GameLiftConfiguration configuration)
         {
             _configuration = configuration;
             _userIdentity = new UserIdentity(new AmazonCognitoIdentityWrapper(configuration.AwsRegion));
             _apiGateway = new ApiGateway(_userIdentity, new JwtTokenExpirationCheck(), new HttpClientWrapper());
```

Comment:
We have a lot of if checks for local/anywhere for switching between userIdentity/apiGateway or mocked responses and the anywhere adapter. Instead consider moving to a strategy pattern.

Here's an example piece of code if we used the interface shared between anywhere adapter and apigateway, and mocked the user identity interface for use with anywhere:
```
private readonly IGameServerAdapter _gameServerAdapter;
private readonly IUserIdentity _userIdentity;

...

_configuration = configuration
// The following could even be moved to a factory at some point with little refactoring needed
if (_configuration.IsGameLiftAnywhere)
{
  _userIdentity = new MockUserIdentity();
  var gameLiftClientWrapper = new AmazonGameLiftWrapper(_configuration.ApiGatewayEndpoint);
  _gameServerAdapter = new AnywhereGameServerAdapter(gameLiftClientWrapper);
}
else
{
  _userIdentity = new UserIdentity(new AmazonCognitoIdentityWrapper(configuration.AwsRegion));
  _gameServerAdapter = new ApiGateway(_userIdentity, new JwtTokenExpirationCheck(), new HttpClientWrapper());
}

...

public virtual SignUpResponse SignUp(string email, string password)
{
  var request = new SignUpRequest()
  {
    ClientId = _configuration.UserPoolClientId,
    Username = email,
    Password = password,
  };
  return _userIdentity.SignUp(request);
}

...

public virtual Task<GetGameConnectionResponse> GetGameConnection(string idToken, string refreshToken)
{
  var request = new GetGameConnectionRequest()
  {
    ClientId = _configuration.UserPoolClientId,
    ApiGatewayEndpoint = _configuration.ApiGatewayEndpoint,
    IdToken = idToken,
    RefreshToken = refreshToken,
  };

  return _gameServerAdapter.GetGameConnection(request);
}
```

Reply:
I agree this is a better method of doing dealing with this. But as we never use the _userIdentity for an anywhere game, and I can see that to create this mock _userIdentity I would be doing a similar thing but in a different place (inside of UserIdentity.cs) and overall just would be overcomplicating the issue. 
I have implemented the pattern for the _gameServerAdapter as that is already setup to be used like that.

Reply:
Thanks for using that pattern with game server adapter, and I'm fine to move on and leave user identity as is.

However just to clarify, the change I'm describing would not touch UserIdentity.cs. Instead you'd create another class MockUserIdentity that implements the interface IUserIdentity and returns OK responses for each call. You'd then create a MockUserIdentity for anywhere configuration instead of the UserIdentity in the constructor (or factory) and we can remove all the extra if conditions on whether we are in anywhere mode.